yarn.config.cjs

```diff
@@ -0,0 +1,273 @@
+ */
+async function expectReadme(workspace, workspaceName) {
+  const readme = await getWorkspaceFile(workspace, 'README.md');
+  if (
+    workspaceName !== 'metamask-module-template' &&
+    readme.includes('## Template Instructions')
+  ) {
+    workspace.error(
+      'The README.md contains template instructions. These instructions should be removed.',
+    );
+  }
+
+  if (!readme.includes(`yarn add @metamask/${workspaceName}`)) {
+    workspace.error(
+      `The README.md does not contain an example of how to install the package using Yarn (\`yarn add @metamask/${workspaceName}\`). Please add an example.`,
+    );
+  }
+
+  if (!readme.includes(`npm install @metamask/${workspaceName}`)) {
+    workspace.error(
+      `The README.md does not contain an example of how to install the package using npm (\`npm install @metamask/${workspaceName}\`). Please add an example.`,
+    );
+  }
+}
+
+/**
+ * Expect that the workspace has a pull_request_template.md file, and that it
+ * is a non-empty string. The pull_request_template.md is expected to:
+ *
+ * - Not contain an examples section (unless the workspace is the module
+ * template itself).
+ *
+ * @param {Workspace} workspace - The workspace to check.
+ * @param {string} workspaceName - The name of the workspace.
+ * @returns {Promise<void>}
+ */
+async function expectPullRequestTemplate(workspace, workspaceName) {
+  if (workspaceName === 'metamask-module-template') {
+    return;
+  }
+
+  const pullRequestTemplate = await getWorkspaceFile(
+    workspace,
+    '.github/PULL_REQUEST_TEMPLATE.md',
+  );
+
+  if (!pullRequestTemplate) {
+    workspace.error(
+      'The pull_request_template.md is missing. This should be added.',
+    );
+  }
+
+  if (pullRequestTemplate.includes('## Examples')) {
+    workspace.error(
+      'The pull_request_template.md contains an examples section. This section should be removed.',
+    );
+  }
+}
+
+/**
+ * Expect that the workspace has a valid `exports` field. The `exports` field
+ * is expected to:
+ *
+ * - Export a `types` entrypoint as the first export, or not at all.
+ *
+ * This is required for proper TypeScript support when using `Node16` (or later)
+ * module resolution.
+ *
+ * @param {Workspace} workspace - The workspace to check.
+ * @returns {void}
+ */
+function expectExports(workspace) {
+  const { exports: manifestExports } = workspace.manifest;
+  Object.entries(manifestExports)
+    .filter(([, exportValue]) => typeof exportValue !== 'string')
+    .forEach(([exportName, exportObject]) => {
+      const keys = Object.keys(exportObject);
+      if (keys.includes('types') && keys[0] !== 'types') {
+        workspace.error(
+          `The "types" export must be the first export in the "exports" field for the export "${exportName}".`,
+        );
+      }
+    });
+}
+
+module.exports = defineConfig({
+  async constraints({ Yarn }) {
+    const workspace = Yarn.workspace();
+    const workspaceName = basename(__dirname);
```

Comment:
Making this dependent on the directory name makes sense, but it also means that this constraint may act differently depending on where it's run. For instance, if a repo is renamed but the developer doesn't know it and they haven't renamed their directory yet then the constraint could suddenly fail for them.

It it possible to assert that the `name` field is present, and then derive the `workspaceName` from that?

Reply:
I've changed it to use `workspace.ident` instead.